gitops-v2/new/push.yaml

```diff
@@ -71,38 +82,85 @@ jobs:
-            if [[ ! -f "${ARTIFACT_PATH}" ]]
-            then
-                echo "Missing artifact file ($ARTIFACT_PATH) from CI pipeline, did you trigger this pipeline manually?"
-                exit 1
-            fi
+
             IMAGE_ID_FULL="${ACR_NAME}.azurecr.io/${IMAGE_PATH_PREFIX}/${IMAGE_ID}"
             az acr login -n "${ACR_NAME}"
-            docker load < $ARTIFACT_PATH
+        condition: ne('${{ parameters.azureSubscription }}', '')
+
+      - task: AmazonWebServices.aws-vsts-tools.AWSShellScript.AWSShellScript@1
+        displayName: "Get ECR login info"
+        inputs:
+          workingDirectory: "."
+          awsCredentials: ${{ parameters.awsServiceConnection }}
+          regionName: ${{ parameters.awsRegion }}
+          scriptType: inline
+          inlineScript: |
+            set -e
+
+            aws ecr describe-repositories --repository-names ${SERVICE_NAME} || aws ecr create-repository --repository-name ${SERVICE_NAME} --image-scanning-configuration scanOnPush=true
+            ECR_URI=$(aws ecr describe-repositories --repository-names ${SERVICE_NAME} | jq -r '.repositories[0].repositoryUri')
+            ECR_PASSWORD=$(aws ecr get-login-password --region ${AWS_REGION})
+            docker login -u AWS -p $ECR_PASSWORD $ECR_URI
+
+            echo "##vso[task.setvariable variable=server]$ECR_URI"
+        env:
+          SERVICE_NAME: $(resources.triggeringAlias)
+        condition: ne('${{ parameters.awsRegion }}', '')
+
+      - bash: |
+          set -e
+          WORKSPACE=$(Pipeline.Workspace)
+          EXTRA_TAG=${{ parameters.extraTag }}
+          SHA_SHORT=$(shaShort)
+          SERVER=$(server)
+          ACR_NAME=${{ parameters.acrName }}
+          IMAGE_PATH_PREFIX=${{ parameters.imagePathPrefix }}
+          IMAGE_ID_FULL="${ACR_NAME}.azurecr.io/${IMAGE_PATH_PREFIX}/${IMAGE_ID}"
+
+          ARTIFACT_PATH="${WORKSPACE}/${SERVICE_NAME}/${ARTIFACT_FILE_NAME}"
+          if [[ ! -f "${ARTIFACT_PATH}" ]]
+          then
+              echo "Missing artifact file ($ARTIFACT_PATH) from CI pipeline, did you trigger this pipeline manually?"
+              exit 1
+          fi
+
+          docker load < $ARTIFACT_PATH
+
+          # Run AWS
+          if [[ ${{ parameters.awsRegion }} != "" ]]
+          then
+            docker tag $(imageId) ${SERVER}:${SHA_SHORT}
+            docker push ${SERVER}:${SHA_SHORT}
+          fi
+
+          # Run Azure
+          if [[ $IMAGE_PATH_PREFIX != "" ]]
+          then
             docker tag $(imageId) "${IMAGE_ID_FULL}"
             docker push "${IMAGE_ID_FULL}"
-            if [[ "${EXTRA_TAG}" != "" ]]
-            then
-              docker tag ${SERVICE_NAME}:${EXTRA_TAG}
-              docker push ${SERVICE_NAME}:${EXTRA_TAG}
-            fi
+          fi
+
+          if [[ "${EXTRA_TAG}" != "" ]]
+          then
+            docker tag ${SERVICE_NAME}:${EXTRA_TAG}
+            docker push ${SERVICE_NAME}:${EXTRA_TAG}
+          fi
+
```

Comment:
I actually don't see how this one works. No registry to push to.. ?

Reply:
Me nether to be honest, I guess it works if you have a repo that is public for everyone to write too :)
I just stayed consistent with the current solution. I will remove them both